File: evidence/generated-systems/angular/fintech-on-angular/src/app/models/Screening.js

// Define collection and schema for Screening
export interface Screening {
    score:
	type : RiskScore
    screenedAt:
	type : DateTime
    KycProfile:
	type : Schema.Types.ObjectId
    Alerts:
 	type : [{ type: Schema.Types.ObjectId, ref: 'ComplianceAlert' }]
    ScreeningType:
 	type : String
    Status:
 	type : String
#
    collection: 'screenings'
}

File: evidence/generated-systems/rubyonrails/fintech-on-rails/fintechonrails/app/models/screening.rb
class Screening < ApplicationRecord
  enum ScreeningType: [:Sanctions, :PEP, :AdverseMedia]
  enum Status: [:Clear, :Review, :Match]


  composed_of :riskScore,
    class_name: "RiskScore",
    mapping: [
      %w[riskScore_value value]
    ]

  composed_of :dateTime,
    class_name: "DateTime",
    mapping: [
      %w[dateTime_value value]
    ]

  has_many :KycProfile, class_name: 'KYCProfile'
  has_many :Alerts, class_name: 'ComplianceAlert'

end
